compressors/integer_coders.py

```diff
@@ -85,3 +86,149 @@ def test_universal_uint_encode_decode():
+
+        q = x // self.M  # quotient
+        r = x % self.M  # remainder
+
+        # encode quotient in unary
+        quotient_bitarray = BitArray(q * "1" + "0")
+        # encode remainder in binary using b bits if r < cutoff,
+        # or else encode r + cutoff using b+1 bits
+        # This is the https://en.wikipedia.org/wiki/Truncated_binary_encoding
+        # For M power of 2 (Rice code, always go with b bits)
+        if self.rice_code or r < self.cutoff:
+            remainder_bitarray = uint_to_bitarray(r, bit_width=self.b)
+        else:
+            remainder_bitarray = uint_to_bitarray(r + self.cutoff, bit_width=self.b + 1)
+
+        return quotient_bitarray + remainder_bitarray
+
+    def encode_block(self, data_block: DataBlock):
+        encoded_bitarray = BitArray("")
+        for s in data_block.data_list:
+            encoded_bitarray += self.encode_symbol(s)
+        return encoded_bitarray
+
+
+class GolombUintDecoder(DataDecoder):
+    """
+    Golomb code with parameter M based on
+    https://en.wikipedia.org/wiki/Golomb_coding#Simple_algorithm:
+    If M is power of 2, we simplify to Rice codes with slight
+    change in logic.
+    """
+
+    def __init__(self, M: int):
+        _initialize_golomb_coder(self, M)
+
+    def decode_symbol(self, encoded_bitarray):
+
+        # initialize num_bits_consumed
+        num_bits_consumed = 0
+
+        # infer the quotient
+        while True:
+            bit = encoded_bitarray[num_bits_consumed]
+            num_bits_consumed += 1
+            if bit == 0:
+                break
+
+        quotient = num_bits_consumed - 1
+
+        # see if next bit is 0 or 1 to figure out if we encoded remainder with b or b+1 bits
+        # For M power of 2 (Rice code, always go with b bits)
+        if self.rice_code or str(encoded_bitarray[num_bits_consumed]) == "0":
+            num_bits_remainder = self.b
+            remainder = bitarray_to_uint(
+                encoded_bitarray[num_bits_consumed : num_bits_consumed + num_bits_remainder]
+            )
+            num_bits_consumed += num_bits_remainder
+        else:
+            num_bits_remainder = self.b + 1
+            remainder = (
+                bitarray_to_uint(
+                    encoded_bitarray[num_bits_consumed : num_bits_consumed + num_bits_remainder]
+                )
+                - self.cutoff
+            )
+            num_bits_consumed += num_bits_remainder
+
+        symbol = self.M * quotient + remainder
+
+        return symbol, num_bits_consumed
+
+    def decode_block(self, bitarray: BitArray):
+        data_list = []
+        num_bits_consumed = 0
+        while num_bits_consumed < len(bitarray):
+            s, num_bits = self.decode_symbol(bitarray[num_bits_consumed:])
+            num_bits_consumed += num_bits
+            data_list.append(s)
+
+        return DataBlock(data_list), num_bits_consumed
+
+
+def _test_golomb_encode_decode_helper(M, data_list, expected_output_bitarray):
```

Comment:
could use the "try_lossless..." utility function

Reply:
Yes, I think it might be in general useful.. can you change ` try_lossless_compression` to return the `encoded_bitarray` (instead of the length). and then we can check outside the function 

```
    is_lossless, encoded_bitarray = try_lossless_compression(data_block, encoder, decoder)
    assert is_lossless
    assert encoded_bitarray == expected_output_bitarray
```

This way, in case we don't want to test the `encoded_bitarray`, that output can be ignored.

Reply:
I use this add-on (suggested by @tpulkit) which makes it quite easy

```
Name: autoDocstring - Python Docstring Generator
Id: njpwerner.autodocstring
Description: Generates python docstrings automatically
Version: 0.6.1
Publisher: Nils Werner
VS Marketplace Link: https://marketplace.visualstudio.com/items?itemName=njpwerner.autodocstring
```